compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -288,14 +290,30 @@ void addGPUTileAndFusePassPipeline(OpPassManager &funcPassManager) {
     funcPassManager.addPass(createCSEPass());
   }
 
-  // Step 2. Tile and fuse tileable ops to threads.
+  // Decompose pack and unpack ops and propagte the resulting reshapes.
+  funcPassManager.addPass(
+      createDecomposePackUnPackOpsPass(/*tileOuterToOne=*/false));
+  funcPassManager.addPass(createPropagateReshapesByExpansionPass());
+  funcPassManager.addPass(createCanonicalizerPass());
+  funcPassManager.addPass(createCSEPass());
+  funcPassManager.addPass(createCleanupBufferAllocViewPass());
```

Comment:
Is it possible to add the reshape tensor interface foldings to `PropagateReshapesByExpansionPass` and avoid having both passes here?

Reply:
Unfortunately the canonicalization passes are still load bearing (to allow DPS to connect in the next pass) so I left those there, but adding the interface foldings let us drop the alloc view pass, which is nice.